concourse/steps/update_component_deps.py

```diff
@@ -377,6 +378,125 @@ def _import_release_notes(
     return release_notes
 
 
+def format_bom_diff(component_diff, include_internal_links=False):
+    changed_components = []
+    added_components = []
+    removed_components = []
+    changed_versions = []
+
+    if (hasattr(component_diff, 'cidentities_only_left') and
+        hasattr(component_diff, 'cidentities_only_right')):
+        left_components = component_diff.cidentities_only_left
+        right_components = component_diff.cidentities_only_right
+
+        for component_identity in left_components:
+            matching_component = next((comp for comp in right_components
+                                       if comp.name == component_identity.name), None)
+            if matching_component:
+                if matching_component.version != component_identity.version:
+                    changed_versions.append(
+                        f'🔄 {component_identity.name}:'
```

Comment:
I would like to avoid emojis in source code, can we rather use codepoints? (e.g. `"\u2764"`)

Reply:
also, I suggest to group components (changed, removed, ...) as objects and only apply (markdown) formatting and the very end

Reply:
@zkdev : well... python3 does allow use of unicode, so I do not see a strong reason not to (see [documentation on unicode](https://docs.python.org/3/howto/unicode.html); we might add an encoding-hint to the top of this module - however, considering utf-8 is the default, it should be safe (and formally correct) to omit it.